File: gloSuite9023/gloSuite/gloCommon/gloUserControlLibrary/UserControls/gloSearchTextBox.vb
Public Class gloSearchTextBox
    Inherits TextBox

    Public Event SearchFired()
    Dim _CurrentTime As Date


    Private Sub oTimer_Tick() Handles oTimer.Tick

        If Me.Text.Trim <> "" Then

            If DateTime.Now.Subtract(_CurrentTime).Milliseconds > 100 Then
                oTimer.Stop()
                ''Added by Mayuri: to set focus on sleected node after entering code
                If Me.Tag = "" Then
                    RaiseEvent SearchFired()
                ElseIf Me.Tag = "True" Then
                    Me.Tag = ""

                End If
            End If
                '


        Else
            oTimer.Stop()
            ''Added by Mayuri: to set focus on sleected node after entering code
            If Me.Tag = "" Then
                RaiseEvent SearchFired()
            ElseIf Me.Tag = "True" Then
                Me.Tag = ""

            End If



        End If


    End Sub

    Private Sub gloSearchTextBox_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles Me.KeyDown
        ''Added on 20140221-To disable enter while searching in textbox
        If e.KeyValue = Keys.Enter Then
            Exit Sub
        End If
        _CurrentTime = DateTime.Now
        oTimer.Stop()
        oTimer.Interval = 700
        oTimer.Enabled = True
    End Sub

    Private Sub gloSearchTextBox_TextChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.TextChanged
       
        If oTimer.Enabled = False Then
            oTimer.Stop()
            oTimer.Enabled = True
        End If
    End Sub

End Class

File: gloSuite9023/gloSuite/gloCommon/gloSnoMed/gloSearchTextBox.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace gloSnoMed
{
    public partial class gloSearchTextBox : TextBox
    {
        public event SearchFiredEventHandler SearchFired;
        public delegate void SearchFiredEventHandler();

        System.DateTime _CurrentTime;

        private void oTimer_Tick(object sender, EventArgs e)
        {

            if (!string.IsNullOrEmpty(this.Text.Trim()))
            {
                if (DateTime.Now.Subtract(_CurrentTime).Milliseconds > 100)
                {
                    oTimer.Stop();
                    if (SearchFired != null)
                    {
                        SearchFired();
                    }
                }

            }
            else
            {
                oTimer.Stop();
                if (SearchFired != null)
                {
                    SearchFired();
                }
            }

        }

        private void gloSearchTextBox_KeyDown(object sender, System.Windows.Forms.KeyEventArgs e)
        {
            //'Added on 20140221-To disable enter while searching in textbox
            if (e.KeyValue  ==(int) Keys.Enter)
            {
                return;
            }
            _CurrentTime = DateTime.Now;
            oTimer.Stop();
            oTimer.Interval = 700;
            oTimer.Enabled = true;
        }

        private void gloSearchTextBox_TextChanged(object sender, System.EventArgs e)
        {
       
            if (oTimer.Enabled == false)
            {
                oTimer.Stop();
                oTimer.Enabled = true;
            }
        }
        public gloSearchTextBox()
        {
            InitializeComponent(); 
            TextChanged += gloSearchTextBox_TextChanged;
            KeyDown += gloSearchTextBox_KeyDown;
        }

       
    }
}
